### PR TITLE
Feature/moar docs

### PR DESCRIPTION
**Pull request by agius**
Add more component docs, add a full-page example to show how `box`, `layout`, and `layout__item` work together.

### CODE REVIEW

**Comment by agius** [comment]
This is like, 90% tab changes. HAML-- :frowning:

**Comment by wnadeau** [comment]
nice! i <3 the new docs.

I'm intrigued by this new YAML documentation pathway. It looks like some of the components use it, but most of them don't. How did you intend to see it used?

**Comment by agius** [comment]
I think we could migrate most of the components to it. Only the ones that have special / unusual layouts would need an actual partial. I just didn't get too far in migrating them over before winter break.

File: app/views/fortitude_rails/static/_component_doc.html.haml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+.layout__item><
+  %h2.text-black-light.xs-beta.xs-inset2= title
+  %hr.text-gray-light
+  %p.xs-ptb1.xs-delta= description
+  %pre
+    %code.scss
+      = code_sample
+
+- if block_given?
```
Comment (wnadeau) [inline app/views/fortitude_rails/static/_component_doc.html.haml]: can you point me to a consumer that is providing a block for this partial? (not sure where to look)

Is this the way you intended for examples to passed along with the accompanying yml attributes?

**Comment by wnadeau** [comment]
I don't see `examples` as an attribute in the yml 

image

so it seems kind of strange that examples might be defined in a different place than their samples.

TBH, my first reaction is mostly "why not just html?",  but I can definitely be sold if I could get a better view of your final picture

**Comment by agius** [comment]
Agreed. I tried 3 or 4 different things, didn't fully decide on one, and didn't migrate everything to it. Out of scope for just adding more docs.

Removed from branch - I'll make a different branch for those experiments.

**Comment by wnadeau** [comment]
:+1: :+1: this is awesome

**Pull request merged**